tests/validator_test.go

```diff
@@ -329,4 +337,611 @@ var _ = Describe("Template validator", func() {
+		)
+
+		BeforeEach(func() {
+			vmi = NewRandomVMIWithBridgeInterface(strategy.GetNamespace())
+			// Create a dummy template to make sure there's no value from an earlier test
+			// And dummy name & namespace for a valid object in Cleanup
+			template = templatev1.Template{}
+			template.Name = "testDummy"
+			template.Namespace = "testDummy"
+		})
+		AfterEach(func() {
+			err := apiClient.Delete(ctx, &template)
+			if !errors.IsNotFound(err) {
+				Expect(err).ToNot(HaveOccurred(), "Failed to delete Template")
+			}
+			err = apiClient.Delete(ctx, vm)
+			if !errors.IsNotFound(err) {
+				Expect(err).ToNot(HaveOccurred(), "Failed to Delete VM")
+			}
+		})
+
+		It("should create VM without template", func() {
+			vm = NewVirtualMachine(vmi)
+			Expect(apiClient.Create(ctx, vm)).ToNot(HaveOccurred(), "Failed to create VM")
+		})
+		It("be created from template with minimal rules", func() {
+			template = TemplateWithMinimalRule()
+			Expect(apiClient.Create(ctx, &template)).ToNot(HaveOccurred(), "Failed to create template: %s", template.Name)
+
+			vm = NewVirtualMachine(vmi)
+			vm.ObjectMeta.Annotations = map[string]string{
+				TemplateNameAnnotation:      template.Name,
+				TemplateNamespaceAnnotation: template.Namespace,
+			}
+			// TODO: Investigate why this test consistently fails without Eventually
+			Eventually(func() error {
+				return apiClient.Create(ctx, vm)
+			}, shortTimeout).Should(BeNil(), "Failed to create VM")
+		})
+		It("[test_id:5033]: Template with validations, VM without validations", func() {
+			template = TemplateWithRules()
+			Expect(apiClient.Create(ctx, &template)).ToNot(HaveOccurred(), "Failed to create template: %s", template.Name)
+
+			vmi = addDomainResourcesToVMI(vmi, 2, "q35", "128M")
+			vm = NewVirtualMachine(vmi)
+			vm.ObjectMeta.Annotations = map[string]string{
+				TemplateNameAnnotation:      template.Name,
+				TemplateNamespaceAnnotation: template.Namespace,
+			}
+			Expect(apiClient.Create(ctx, vm)).ToNot(HaveOccurred(), "Failed to create VM")
+		})
+		It("[test_id:2960] Negative test - Create a VM with machine type violation", func() {
+			template = TemplateWithRules()
+			Expect(apiClient.Create(ctx, &template)).ToNot(HaveOccurred(), "Failed to create template: %s", template.Name)
+
+			// set value unfulfilling validation
+			vmi = addDomainResourcesToVMI(vmi, 2, "test", "128M")
+			vm = NewVirtualMachine(vmi)
+			vm.ObjectMeta.Annotations = map[string]string{
+				TemplateNameAnnotation:      template.Name,
+				TemplateNamespaceAnnotation: template.Namespace,
+			}
+			Expect(errors.IsInvalid(apiClient.Create(ctx, vm))).To(BeTrue(), "Should match error type because of unfulfilled validations")
+		})
+		It("test with template optional rules unfulfilled", func() {
+			template = TemplateWithRulesOptional()
+			Expect(apiClient.Create(ctx, &template)).ToNot(HaveOccurred(), "Failed to create template: %s", template.Name)
+
+			vmi = addDomainResourcesToVMI(vmi, 0, "q35", "128M")
```

Comment:
The VM in the original test does not have CPU information. Please remove it from `vmi`: `vmi.Spec.Domain.CPU = nil`. See my other comment for explanation why.

Since the rules are optional, the validation should pass even if no CPU info is provided.